src/H5WebViewer.ts

```diff
@@ -1,34 +1,24 @@
 import {
-  CancellationToken,
-  CustomDocumentOpenContext,
+  CustomDocument,
   CustomReadonlyEditorProvider,
   ExtensionContext,
   Uri,
   Webview,
   WebviewPanel,
-  workspace,
 } from 'vscode';
-import HDF5Document from './HDF5Document';
 import { join, basename } from 'path';
 
 export default class H5WebViewer
-  implements CustomReadonlyEditorProvider<HDF5Document>
+  implements CustomReadonlyEditorProvider<CustomDocument>
 {
   public constructor(private readonly context: ExtensionContext) {}
 
-  public async openCustomDocument(
-    uri: Uri,
-    openContext: CustomDocumentOpenContext
-  ): Promise<HDF5Document> {
-    const { backupId } = openContext;
-    const fileUri = typeof backupId === 'string' ? Uri.parse(backupId) : uri;
-
-    const buffer = await workspace.fs.readFile(fileUri);
-    return new HDF5Document(uri, buffer);
+  public async openCustomDocument(uri: Uri): Promise<CustomDocument> {
+    return { uri, dispose: () => {} };
```

Comment:
I'm actually removing the concept of `HDF5Document` entirely, since we really just need the URI of the file... This is what they do in the built-in image preview: https://github.com/microsoft/vscode/blob/24c814dd015dbbfe1542662413ddadc5197902d4/extensions/image-preview/src/preview.ts#L30

Reply:
Good find :detective: